### ISSUE TITLE
Command Block /topoislands home not functioning

### ISSUE THREAD

**Issue by Lightshardz** [sam-kirby/topoislands#1]
Would like to set up a "Lobby" in SkyFactory One 1.16.5 where the new players will spawn in at World Spawn and land on a command block that "/execute @p ~ ~ ~ /topoislands home". This would a) teleport the player to their assigned island, if existing or b) create a new island for said player and teleport them to it.
The command /topoislands home works if a player has an island assigned.
The  command /topoislands new <PLAYER_NAME> works if an admin is actively online and can execute for player.
"/topoislands home" from a command block requires a player to run the command, hence trying the execute command.

**Comment by Lightshardz** [comment]
"execute run topoislands new @p" is not causing anything to occur. Nothing is going to the console or log file. "execute /topoislands new @p" throws an invalid syntax.
"execute run topoislands home" throws the "A player is required to use this command"
"execute as @p run topoislands home" does work, but it is teleporting me directly back to Spawn rather than my Island ID of 2 (I was originally 1, somewhere in this mix it generated a new one for me)
"execute as @p run topoislands new @p" is not working as above without the as command.

**Comment by sam-kirby** [comment]
Oh, and I should make clear - the home command teleports the player that runs it to their currently assigned island. If they have no island assigned it teleports them to spawn. It does not create an island under any circumstance and cannot meaningully be run by a command block.

**Comment by Lightshardz** [comment]
Got it! 
So, for clarification. I was on an Island (ID 2), I used "/topoislands Spawn" to go back to my "lobby" where I was messing with my command block. When I had the Command Block execute the action "execute as @p run /topoislands home" it dropped me right back on Spawn instead of back to my Island ID 2. I could have used your assign function to put me back to 2, but it had unlinked me as Island 2. That also happened to me when messing with the command block from my Island ID 1. So while I was at Spawn and using /home and staying there, I used New again and it made 3, then I used new again and it made 4. That all makes sense, I understand that. But going to spawn, then the command block using the /home I think is what was dropping me back on Home with no Island ID set.
